clients/src/main/java/org/apache/kafka/common/record/FileRecords.java

```diff
@@ -427,7 +442,7 @@ public static FileRecords open(File file,
                                    boolean preallocate) throws IOException {
         FileChannel channel = openChannel(file, mutable, fileAlreadyExists, initFileSize, preallocate);
         int end = (!fileAlreadyExists && preallocate) ? 0 : Integer.MAX_VALUE;
-        return new FileRecords(file, channel, 0, end, false);
+        return new FileRecords(file, channel, 0, end, false, mutable);
```

Comment:
This is ok but not the most accurate. We only need to set the flush flag to true if it's mutable and log recovery is needed.

Reply:
fixed with passing in `hadCleanShutdown`